components/director/internal/domain/apptemplate/resolver.go

```diff
@@ -449,7 +448,7 @@ func (r *Resolver) UpdateApplicationTemplate(ctx context.Context, id string, in
 		return nil, err
 	}
 	if isSelfRegFlow {
-		if err := validateAppTemplateForSelfReg(convertedIn.Placeholders); err != nil {
+		if err := validateAppTemplateForSelfReg(convertedIn.ApplicationInputJSON); err != nil {
```

Comment:
use "in.ApplicationInput" in order to avoid casting like - appName.(string), labels.(map[string]interface{})...

Reply:
that's a good idea, yep 👍